src/plugins/administrative_templates/ui/presentationbuilder.cpp

```diff
@@ -107,6 +107,107 @@ QHBoxLayout *createCaptions()
     return horizontalLayout;
 }
 
+QMap<std::string, QString> loadListFromRegistry(AbstractRegistrySource &m_source, const std::string &key, const std::string &prefix)
+{
+    QMap<std::string, QString> items;
+    std::vector<std::string> valueNames = m_source.getValueNames(key);
+    
+    // finding and erase **delvals.
+    for (auto it = valueNames.begin(); it != valueNames.end(); ++it) {
+        if (QString::fromStdString(*it).compare("**delvals.", Qt::CaseInsensitive) == 0) {
+            it = valueNames.erase(it);
+
+            // std::vecotor<T>::erase() returning iterator that pointing 
+            // to the next element after erased
+            --it; 
+            break;
+        }
+    }
+    
+    for (auto &valueName : valueNames) {
+        items[valueName] = 
+                m_source.getValue(key, valueName).value<QString>();
+    }
+    return items;
+}
+
+void cleanUpListInRegistry(AbstractRegistrySource &m_source, const std::string &key)
+{
+    std::vector<std::string> valueNames = m_source.getValueNames(key);
+    
+    // just clean-up all values on this keypath
+    for (auto &value : valueNames) {
+        m_source.clearValue(key, value);
+    }
+}
+
+void cleanUpListInRegistry(AbstractRegistrySource &m_source, const std::string &key, const std::string &prefix)
+{
+    // small optimization
+    if (prefix.size() == 0)
+    {
+        cleanUpListInRegistry(m_source, key);
+    }
+
+    QString _prefix = QString::fromStdString(prefix);
+
+    std::vector<std::string> valueNames = m_source.getValueNames(key);
+    
+    // clean-up all values that contain `prefix` prefix (case-insensitive)
+    for (auto &value : valueNames) {
+        if (value.size() > prefix.size() && 
+            QString::fromUtf8(value.c_str(), prefix.size()).compare(_prefix, Qt::CaseInsensitive))
```

Comment:
`QString::compare()` will return 0 if `_prefix` and `value` are the same. We need to add `== 0` here. We should avoid mixing `prefix` and `_prefix` in same condition it is difficult to read.